Weather_Display_Matrix/weather_display_matrix.py

```diff
@@ -0,0 +1,88 @@
+# Matrix Weather display
+# For Metro M4 Airlift with RGB Matrix shield, 64 x 32 RGB LED Matrix display
+
+"""
+This example queries the Open Weather Maps site API to find out the current
+weather for your location... and display it on a screen!
+if you can find something that spits out JSON data, we can display it
+"""
+import time
+import board
+from digitalio import DigitalInOut, Direction, Pull
+from adafruit_matrixportal.network import Network
+from adafruit_matrixportal.matrix import Matrix
+import openweather_graphics  # pylint: disable=wrong-import-position
+
+# Get wifi details and more from a secrets.py file
+try:
+    from secrets import secrets
+except ImportError:
+    print("WiFi secrets are kept in secrets.py, please add them there!")
+    raise
+
+jumper = DigitalInOut(board.D12)
+jumper.direction = Direction.INPUT
+jumper.pull = Pull.UP
+
+if jumper.value:
+    UNITS = "metric"  # can pick 'imperial' or 'metric' as part of URL query
+    print("Jumper set to metric")
+else:
+    UNITS = "imperial"
+    print("Jumper set to imperial")
+
+# Use cityname, country code where countrycode is ISO3166 format.
+# E.g. "New York, US" or "London, GB"
+LOCATION = "Los Angeles, US"
+print("Getting weather for {}".format(LOCATION))
+# Set up from where we'll be fetching data
+DATA_SOURCE = (
+    "http://api.openweathermap.org/data/2.5/weather?q=" + LOCATION + "&units=" + UNITS
+)
+DATA_SOURCE += "&appid=" + secrets["openweather_token"]
+# You'll need to get a token from openweather.org, looks like 'b6907d289e10d714a6e88b30761fae22'
+# it goes in your secrets.py file on a line such as:
+# 'openweather_token' : 'your_big_humongous_gigantor_token',
+DATA_LOCATION = []
+SCROLL_HOLD_TIME = 0  # set this to hold each line before finishing scroll
+
+# --- Display setup ---
+matrix = Matrix()
+network = Network(status_neopixel=board.NEOPIXEL, debug=True)
+if UNITS == "imperial":
+    gfx = openweather_graphics.OpenWeather_Graphics(
+        matrix.display, am_pm=True, units="imperial"
+    )
+elif UNITS == "metric":
+    gfx = openweather_graphics.OpenWeather_Graphics(
+        matrix.display, am_pm=True, units="metric"
+    )
```

Comment:
This could be reduced to:
```
if UNITS == "imperial" or UNITS == "metric":
    gfx = openweather_graphics.OpenWeather_Graphics(
        matrix.display, am_pm=True, units=UNITS
    )

Reply:
Done!